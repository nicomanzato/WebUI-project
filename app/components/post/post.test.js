import React from 'react'
import Post from './post'
import PostMock2 from './../../store/post/mock/postMock2'
import { shallow, mount, ReactWrapper } from 'enzyme'

describe('post component', () => {

  it('should render correctly', () => {
    const wrapper = shallow(
      <Post item={PostMock2} />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should call onProfilePicPress when pressed', () => {

    const mockFn = jest.fn().mockName('mockedFunction');

    const wrapper = mount(
      <Post
        item={PostMock2}
        onProfilePicPress={() => mockFn()}
      />
    );

    const post = wrapper.find('TouchableHighlight').first();
    post.props().onPress();

    expect(mockFn).toHaveBeenCalled();
  });

});
