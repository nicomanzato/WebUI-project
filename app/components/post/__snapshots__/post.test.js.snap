// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`post component should render correctly 1`] = `
<Component
  style={
    Object {
      "borderBottomWidth": 1,
      "borderColor": "#ccc",
      "padding": 10,
    }
  }
>
  <Component
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <PostProfilePicture
      onProfilePicPress={[Function]}
      user={
        Object {
          "contributors_enabled": false,
          "created_at": "Fri Aug 10 15:14:11 +0000 2007",
          "default_profile": false,
          "default_profile_image": false,
          "description": "Noticias de Argentina y el mundo las 24 horas. Información de último momento, opinión, análisis e informes especiales.",
          "entities": Object {
            "description": Object {
              "urls": Array [],
            },
            "url": Object {
              "urls": Array [
                Object {
                  "display_url": "clarin.com",
                  "expanded_url": "http://www.clarin.com",
                  "indices": Array [
                    0,
                    22,
                  ],
                  "url": "http://t.co/1RuczZo0FK",
                },
              ],
            },
          },
          "favourites_count": 391,
          "follow_request_sent": false,
          "followers_count": 2810398,
          "following": true,
          "friends_count": 265,
          "geo_enabled": true,
          "has_extended_profile": false,
          "id": 8105922,
          "id_str": "8105922",
          "is_translation_enabled": true,
          "is_translator": false,
          "lang": "es",
          "listed_count": 10835,
          "location": "Buenos Aires, Argentina",
          "name": "Clarín",
          "notifications": false,
          "profile_background_color": "131516",
          "profile_background_image_url": "http://abs.twimg.com/images/themes/theme14/bg.gif",
          "profile_background_image_url_https": "https://abs.twimg.com/images/themes/theme14/bg.gif",
          "profile_background_tile": true,
          "profile_banner_url": "https://pbs.twimg.com/profile_banners/8105922/1545358874",
          "profile_image_url": "http://pbs.twimg.com/profile_images/1053387941752369152/r4pGBF3K_normal.jpg",
          "profile_image_url_https": "https://pbs.twimg.com/profile_images/1053387941752369152/r4pGBF3K_normal.jpg",
          "profile_link_color": "FF0324",
          "profile_sidebar_border_color": "FFFFFF",
          "profile_sidebar_fill_color": "EFEFEF",
          "profile_text_color": "333333",
          "profile_use_background_image": true,
          "protected": false,
          "screen_name": "clarincom",
          "statuses_count": 301824,
          "time_zone": null,
          "translator_type": "none",
          "url": "http://t.co/1RuczZo0FK",
          "utc_offset": null,
          "verified": false,
        }
      }
    />
    <Component
      style={
        Object {
          "flex": 0.8,
        }
      }
    >
      <PostUserInformation
        user={
          Object {
            "contributors_enabled": false,
            "created_at": "Fri Aug 10 15:14:11 +0000 2007",
            "default_profile": false,
            "default_profile_image": false,
            "description": "Noticias de Argentina y el mundo las 24 horas. Información de último momento, opinión, análisis e informes especiales.",
            "entities": Object {
              "description": Object {
                "urls": Array [],
              },
              "url": Object {
                "urls": Array [
                  Object {
                    "display_url": "clarin.com",
                    "expanded_url": "http://www.clarin.com",
                    "indices": Array [
                      0,
                      22,
                    ],
                    "url": "http://t.co/1RuczZo0FK",
                  },
                ],
              },
            },
            "favourites_count": 391,
            "follow_request_sent": false,
            "followers_count": 2810398,
            "following": true,
            "friends_count": 265,
            "geo_enabled": true,
            "has_extended_profile": false,
            "id": 8105922,
            "id_str": "8105922",
            "is_translation_enabled": true,
            "is_translator": false,
            "lang": "es",
            "listed_count": 10835,
            "location": "Buenos Aires, Argentina",
            "name": "Clarín",
            "notifications": false,
            "profile_background_color": "131516",
            "profile_background_image_url": "http://abs.twimg.com/images/themes/theme14/bg.gif",
            "profile_background_image_url_https": "https://abs.twimg.com/images/themes/theme14/bg.gif",
            "profile_background_tile": true,
            "profile_banner_url": "https://pbs.twimg.com/profile_banners/8105922/1545358874",
            "profile_image_url": "http://pbs.twimg.com/profile_images/1053387941752369152/r4pGBF3K_normal.jpg",
            "profile_image_url_https": "https://pbs.twimg.com/profile_images/1053387941752369152/r4pGBF3K_normal.jpg",
            "profile_link_color": "FF0324",
            "profile_sidebar_border_color": "FFFFFF",
            "profile_sidebar_fill_color": "EFEFEF",
            "profile_text_color": "333333",
            "profile_use_background_image": true,
            "protected": false,
            "screen_name": "clarincom",
            "statuses_count": 301824,
            "time_zone": null,
            "translator_type": "none",
            "url": "http://t.co/1RuczZo0FK",
            "utc_offset": null,
            "verified": false,
          }
        }
      />
      <PostContent
        post={
          Object {
            "contributors": null,
            "coordinates": null,
            "created_at": "Fri Dec 21 12:52:02 +0000 2018",
            "entities": Object {
              "hashtags": Array [],
              "media": Array [
                Object {
                  "display_url": "pic.twitter.com/GuK1rpkdeJ",
                  "expanded_url": "https://twitter.com/clarincom/status/1076097967147360257/photo/1",
                  "id": 1076097964253298700,
                  "id_str": "1076097964253298688",
                  "indices": Array [
                    100,
                    123,
                  ],
                  "media_url": "http://pbs.twimg.com/media/Du8RZabWwAA5fiX.jpg",
                  "media_url_https": "https://pbs.twimg.com/media/Du8RZabWwAA5fiX.jpg",
                  "sizes": Object {
                    "large": Object {
                      "h": 630,
                      "resize": "fit",
                      "w": 1200,
                    },
                    "medium": Object {
                      "h": 630,
                      "resize": "fit",
                      "w": 1200,
                    },
                    "small": Object {
                      "h": 357,
                      "resize": "fit",
                      "w": 680,
                    },
                    "thumb": Object {
                      "h": 150,
                      "resize": "crop",
                      "w": 150,
                    },
                  },
                  "type": "photo",
                  "url": "https://t.co/GuK1rpkdeJ",
                },
              ],
              "symbols": Array [],
              "urls": Array [
                Object {
                  "display_url": "clarin.com/espectaculos/m…",
                  "expanded_url": "https://www.clarin.com/espectaculos/microteatro-verano-agenda-cargada-pasar-meses-calor_0_Qh15Av7jj.html?utm_term=Autofeed&utm_medium=Social&utm_source=Twitter#Echobox=1545395977",
                  "indices": Array [
                    76,
                    99,
                  ],
                  "url": "https://t.co/3V4dX1OSiJ",
                },
              ],
              "user_mentions": Array [],
            },
            "extended_entities": Object {
              "media": Array [
                Object {
                  "display_url": "pic.twitter.com/GuK1rpkdeJ",
                  "expanded_url": "https://twitter.com/clarincom/status/1076097967147360257/photo/1",
                  "id": 1076097964253298700,
                  "id_str": "1076097964253298688",
                  "indices": Array [
                    100,
                    123,
                  ],
                  "media_url": "http://pbs.twimg.com/media/Du8RZabWwAA5fiX.jpg",
                  "media_url_https": "https://pbs.twimg.com/media/Du8RZabWwAA5fiX.jpg",
                  "sizes": Object {
                    "large": Object {
                      "h": 630,
                      "resize": "fit",
                      "w": 1200,
                    },
                    "medium": Object {
                      "h": 630,
                      "resize": "fit",
                      "w": 1200,
                    },
                    "small": Object {
                      "h": 357,
                      "resize": "fit",
                      "w": 680,
                    },
                    "thumb": Object {
                      "h": 150,
                      "resize": "crop",
                      "w": 150,
                    },
                  },
                  "type": "photo",
                  "url": "https://t.co/GuK1rpkdeJ",
                },
              ],
            },
            "favorite_count": 0,
            "favorited": false,
            "geo": null,
            "id": 1076097967147360300,
            "id_str": "1076097967147360257",
            "in_reply_to_screen_name": null,
            "in_reply_to_status_id": null,
            "in_reply_to_status_id_str": null,
            "in_reply_to_user_id": null,
            "in_reply_to_user_id_str": null,
            "is_quote_status": false,
            "lang": "es",
            "place": null,
            "possibly_sensitive": false,
            "possibly_sensitive_appealable": false,
            "retweet_count": 0,
            "retweeted": false,
            "source": "<a href=\\"https://www.echobox.com\\" rel=\\"nofollow\\">Echobox Social</a>",
            "text": "Microteatro de verano, una agenda cargada para pasar los meses de más calor https://t.co/3V4dX1OSiJ https://t.co/GuK1rpkdeJ",
            "truncated": false,
            "user": Object {
              "contributors_enabled": false,
              "created_at": "Fri Aug 10 15:14:11 +0000 2007",
              "default_profile": false,
              "default_profile_image": false,
              "description": "Noticias de Argentina y el mundo las 24 horas. Información de último momento, opinión, análisis e informes especiales.",
              "entities": Object {
                "description": Object {
                  "urls": Array [],
                },
                "url": Object {
                  "urls": Array [
                    Object {
                      "display_url": "clarin.com",
                      "expanded_url": "http://www.clarin.com",
                      "indices": Array [
                        0,
                        22,
                      ],
                      "url": "http://t.co/1RuczZo0FK",
                    },
                  ],
                },
              },
              "favourites_count": 391,
              "follow_request_sent": false,
              "followers_count": 2810398,
              "following": true,
              "friends_count": 265,
              "geo_enabled": true,
              "has_extended_profile": false,
              "id": 8105922,
              "id_str": "8105922",
              "is_translation_enabled": true,
              "is_translator": false,
              "lang": "es",
              "listed_count": 10835,
              "location": "Buenos Aires, Argentina",
              "name": "Clarín",
              "notifications": false,
              "profile_background_color": "131516",
              "profile_background_image_url": "http://abs.twimg.com/images/themes/theme14/bg.gif",
              "profile_background_image_url_https": "https://abs.twimg.com/images/themes/theme14/bg.gif",
              "profile_background_tile": true,
              "profile_banner_url": "https://pbs.twimg.com/profile_banners/8105922/1545358874",
              "profile_image_url": "http://pbs.twimg.com/profile_images/1053387941752369152/r4pGBF3K_normal.jpg",
              "profile_image_url_https": "https://pbs.twimg.com/profile_images/1053387941752369152/r4pGBF3K_normal.jpg",
              "profile_link_color": "FF0324",
              "profile_sidebar_border_color": "FFFFFF",
              "profile_sidebar_fill_color": "EFEFEF",
              "profile_text_color": "333333",
              "profile_use_background_image": true,
              "protected": false,
              "screen_name": "clarincom",
              "statuses_count": 301824,
              "time_zone": null,
              "translator_type": "none",
              "url": "http://t.co/1RuczZo0FK",
              "utc_offset": null,
              "verified": false,
            },
          }
        }
      />
      <PostMedia
        post={
          Object {
            "contributors": null,
            "coordinates": null,
            "created_at": "Fri Dec 21 12:52:02 +0000 2018",
            "entities": Object {
              "hashtags": Array [],
              "media": Array [
                Object {
                  "display_url": "pic.twitter.com/GuK1rpkdeJ",
                  "expanded_url": "https://twitter.com/clarincom/status/1076097967147360257/photo/1",
                  "id": 1076097964253298700,
                  "id_str": "1076097964253298688",
                  "indices": Array [
                    100,
                    123,
                  ],
                  "media_url": "http://pbs.twimg.com/media/Du8RZabWwAA5fiX.jpg",
                  "media_url_https": "https://pbs.twimg.com/media/Du8RZabWwAA5fiX.jpg",
                  "sizes": Object {
                    "large": Object {
                      "h": 630,
                      "resize": "fit",
                      "w": 1200,
                    },
                    "medium": Object {
                      "h": 630,
                      "resize": "fit",
                      "w": 1200,
                    },
                    "small": Object {
                      "h": 357,
                      "resize": "fit",
                      "w": 680,
                    },
                    "thumb": Object {
                      "h": 150,
                      "resize": "crop",
                      "w": 150,
                    },
                  },
                  "type": "photo",
                  "url": "https://t.co/GuK1rpkdeJ",
                },
              ],
              "symbols": Array [],
              "urls": Array [
                Object {
                  "display_url": "clarin.com/espectaculos/m…",
                  "expanded_url": "https://www.clarin.com/espectaculos/microteatro-verano-agenda-cargada-pasar-meses-calor_0_Qh15Av7jj.html?utm_term=Autofeed&utm_medium=Social&utm_source=Twitter#Echobox=1545395977",
                  "indices": Array [
                    76,
                    99,
                  ],
                  "url": "https://t.co/3V4dX1OSiJ",
                },
              ],
              "user_mentions": Array [],
            },
            "extended_entities": Object {
              "media": Array [
                Object {
                  "display_url": "pic.twitter.com/GuK1rpkdeJ",
                  "expanded_url": "https://twitter.com/clarincom/status/1076097967147360257/photo/1",
                  "id": 1076097964253298700,
                  "id_str": "1076097964253298688",
                  "indices": Array [
                    100,
                    123,
                  ],
                  "media_url": "http://pbs.twimg.com/media/Du8RZabWwAA5fiX.jpg",
                  "media_url_https": "https://pbs.twimg.com/media/Du8RZabWwAA5fiX.jpg",
                  "sizes": Object {
                    "large": Object {
                      "h": 630,
                      "resize": "fit",
                      "w": 1200,
                    },
                    "medium": Object {
                      "h": 630,
                      "resize": "fit",
                      "w": 1200,
                    },
                    "small": Object {
                      "h": 357,
                      "resize": "fit",
                      "w": 680,
                    },
                    "thumb": Object {
                      "h": 150,
                      "resize": "crop",
                      "w": 150,
                    },
                  },
                  "type": "photo",
                  "url": "https://t.co/GuK1rpkdeJ",
                },
              ],
            },
            "favorite_count": 0,
            "favorited": false,
            "geo": null,
            "id": 1076097967147360300,
            "id_str": "1076097967147360257",
            "in_reply_to_screen_name": null,
            "in_reply_to_status_id": null,
            "in_reply_to_status_id_str": null,
            "in_reply_to_user_id": null,
            "in_reply_to_user_id_str": null,
            "is_quote_status": false,
            "lang": "es",
            "place": null,
            "possibly_sensitive": false,
            "possibly_sensitive_appealable": false,
            "retweet_count": 0,
            "retweeted": false,
            "source": "<a href=\\"https://www.echobox.com\\" rel=\\"nofollow\\">Echobox Social</a>",
            "text": "Microteatro de verano, una agenda cargada para pasar los meses de más calor https://t.co/3V4dX1OSiJ https://t.co/GuK1rpkdeJ",
            "truncated": false,
            "user": Object {
              "contributors_enabled": false,
              "created_at": "Fri Aug 10 15:14:11 +0000 2007",
              "default_profile": false,
              "default_profile_image": false,
              "description": "Noticias de Argentina y el mundo las 24 horas. Información de último momento, opinión, análisis e informes especiales.",
              "entities": Object {
                "description": Object {
                  "urls": Array [],
                },
                "url": Object {
                  "urls": Array [
                    Object {
                      "display_url": "clarin.com",
                      "expanded_url": "http://www.clarin.com",
                      "indices": Array [
                        0,
                        22,
                      ],
                      "url": "http://t.co/1RuczZo0FK",
                    },
                  ],
                },
              },
              "favourites_count": 391,
              "follow_request_sent": false,
              "followers_count": 2810398,
              "following": true,
              "friends_count": 265,
              "geo_enabled": true,
              "has_extended_profile": false,
              "id": 8105922,
              "id_str": "8105922",
              "is_translation_enabled": true,
              "is_translator": false,
              "lang": "es",
              "listed_count": 10835,
              "location": "Buenos Aires, Argentina",
              "name": "Clarín",
              "notifications": false,
              "profile_background_color": "131516",
              "profile_background_image_url": "http://abs.twimg.com/images/themes/theme14/bg.gif",
              "profile_background_image_url_https": "https://abs.twimg.com/images/themes/theme14/bg.gif",
              "profile_background_tile": true,
              "profile_banner_url": "https://pbs.twimg.com/profile_banners/8105922/1545358874",
              "profile_image_url": "http://pbs.twimg.com/profile_images/1053387941752369152/r4pGBF3K_normal.jpg",
              "profile_image_url_https": "https://pbs.twimg.com/profile_images/1053387941752369152/r4pGBF3K_normal.jpg",
              "profile_link_color": "FF0324",
              "profile_sidebar_border_color": "FFFFFF",
              "profile_sidebar_fill_color": "EFEFEF",
              "profile_text_color": "333333",
              "profile_use_background_image": true,
              "protected": false,
              "screen_name": "clarincom",
              "statuses_count": 301824,
              "time_zone": null,
              "translator_type": "none",
              "url": "http://t.co/1RuczZo0FK",
              "utc_offset": null,
              "verified": false,
            },
          }
        }
      />
      <SocialInteraction
        favoriteCount={0}
        retweetCount={0}
      />
    </Component>
  </Component>
</Component>
`;
